example-l2/src/executor.rs

```diff
@@ -137,12 +136,18 @@ pub async fn run_executor(opt: &ExecutorOptions, state: Arc<RwLock<State>>) {
 
             assert_eq!(derived_block_comm, block_commitment);
 
-            proofs.push(state.execute_block(nmt_root, namespace_proof).await);
+            proofs.push(
+                state
+                    .write()
+                    .await
+                    .execute_block(nmt_root, namespace_proof)
+                    .await,
+            );
         }
 
         // Compute an aggregate proof.
         let proof = BatchProof::generate(&proofs).expect("Error generating batch proof");
-        let state_comm = commitment_to_u256(state.commit());
+        let state_comm = commitment_to_u256(state.read().await.commit());
```

Comment:
If something else was modifying `state` then I think the commitment and proof could go out of sync. However I think wouldn't be an issue for the current example rollup.